plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/PlayAssetDelivery.java

```diff
@@ -201,13 +201,24 @@ public Dictionary getAssetLocation(String packName, String assetPath) {
    */
   public void fetch(String[] packNamesArray, int signalID) {
     List<String> packNames = Arrays.asList(packNamesArray);
-
     OnSuccessListener<AssetPackStates> fetchSuccessListener =
-        result ->
-            emitSignalWrapper(
-                FETCH_SUCCESS,
-                PlayAssetDeliveryUtils.convertAssetPackStatesToDictionary(result),
-                signalID);
+        result -> {
+          // Handles the edge case where the app is paused immediately after we start this
+          // fetch request. In this case we need to manually add the packNames to
+          // ongoingAssetPackRequests so forceAssetPackStateUpdate() will call
+          // getPackStates() for these packNames. In this implementation, it is possible
+          // to have an extreme edge case where the assetPackState reached terminal state
+          // in globalListener before fetchSuccess is called. This will result in
+          // packName to be added back again to the ongoingAssetPackRequests set and
+          // duplicate stateUpdated signal emitted by forceAssetPackStateUpdate(). Hence
+          // we are using emitNonDuplicateStateUpdatedSignal() to filter out these
+          // duplicate signals.
+          stateUpdateManager.ongoingAssetPackRequests().addAll(result.packStates().keySet());
```

Comment:
I think it's better to add a method to stateUpdateManager to add a list of packStates rather than modifying the underlying ongoingAssetPackRequests contents directly.

Reply:
Added the method of ``` public void joinOngoingAssetPackRequests(Set<String> newOngoingAssetPackRequests) ``` instead of modifying ongoingAssetPackRequests contents directly.